cmd/cephcsi.go

```diff
@@ -55,6 +55,9 @@ func init() {
 		" instances, when sharing Ceph clusters across CSI instances for provisioning")
 	flag.StringVar(&conf.MetadataStorage, "metadatastorage", "", "metadata persistence method [node|k8s_configmap]")
 	flag.StringVar(&conf.PluginPath, "pluginpath", "/var/lib/kubelet/plugins/", "the location of cephcsi plugin")
+	flag.StringVar(&conf.VolumeNamingPrefix, "volume-name-prefix", "", "override the default volume name prefix")
```

Comment:
the custom volume name prefix or snapshot name prefix should come from `storage-class` or `snapshot class`

Reply:
Yeah, agree to @Madhu-1's comment.. So that its kind of a grouping which admin can apply to filter the volumes from SC .etc

Reply:
I think I almost have this figured out, except for the 2 cases in `nodeserver.go` where I don't have enough context for this:
there is a code-path from `NodeStageVolume` and `NodeExpandVolume` (via `getVolumeName`) to `GetNameForUUID`. It makes sense to pass the configured name prefix through that path and this works in the case of ControllerServer's `CreateVolume` etc. These types of request contain the parameters that are accessible through `GetParameters()` but `NodeStageVolumeRequest` and `NodeExpandVolumeRequest` don't have these parameters accessible. 

How do you suggest this should be handled?

Reply:
Perhaps, at that point, I can just get the volume name from the omap key. I’ll see if I can make that work!